tests/tests_app/core/test_lightning_work.py

```diff
@@ -347,3 +347,38 @@ def on_before_run_once(self):
 def test_lightning_app_with_work():
     app = LightningTestAppWithWork(WorkCounter())
     MultiProcessRuntime(app, start_server=False).dispatch()
+
+
+class WorkStart(LightningWork):
+    def __init__(self):
+        super().__init__(parallel=True)
```

Comment:
let's add tests to cover all combinations of semantics (minus `parallel=False` `caching=False`)

Reply:
Done !